_build/lexicon/checklexicon.php

```diff
@@ -0,0 +1,565 @@
+            'message' => $msg
+        );
+
+    }
+
+
+    /**
+     * Load package lexicons
+     *
+     * @return bool|array
+     */
+    private function loadLexicons()
+    {
+        $_lang = array();
+
+        if (file_exists($this->lexiconPath . $this->language . '/')) {
+            $iterator = new \DirectoryIterator($this->lexiconPath . $this->language . '/');
+            foreach ($iterator as $path => $current) {
+                if (strpos($current->getFilename(), 'inc.php') !== false) {
+                    try {
+                        include $current->getRealPath();
+                    } catch (Exception $e) {
+                        $this->invalidLexicons[] = $current->getFilename();
+                    }
+                }
+            }
+        } else {
+            return 'Could not load the lexicons in the language folder "' . $this->lexiconPath . $this->language . '/' . '"!';
+        }
+
+        if (file_exists($this->setupLexiconPath . $this->language . '/')) {
+            $iterator = new \DirectoryIterator($this->setupLexiconPath . $this->language . '/');
+            foreach ($iterator as $path => $current) {
+                if (strpos($current->getFilename(), 'inc.php') !== false) {
+                    try {
+                        include $current->getRealPath();
+                    } catch (Exception $e) {
+                        $this->invalidLexicons[] = $current->getFilename();
+                    }
+                }
+            }
+        } else {
+            return 'Could not load the lexicons in the setup language folder "' . $this->setupLexiconPath . $this->language . '/' . '"!';
+        }
+
+        return $_lang;
+    }
+
+    /**
+     * Add used lexicon keys
+     */
+    private function addKeys()
+    {
+        $directory = new \RecursiveDirectoryIterator($this->scanPath, \RecursiveDirectoryIterator::SKIP_DOTS);
+        $filter = new \RecursiveCallbackFilterIterator($directory, function ($current, $key, $iterator) {
+            /** @var \RecursiveDirectoryIterator $current */
+            if ($current->getFilename()[0] === '.') {
+                return false;
+            }
+            if ($current->isDir()) {
+                return !in_array($current->getFilename(), $this->excludedFolders);
+            } else {
+                $pathinfo = pathinfo($current->getFilename());
+                return ($current->isFile() && isset($pathinfo['extension']) && (
+                        $pathinfo['extension'] == 'php' ||
+                        $pathinfo['extension'] == 'js' ||
+                        $pathinfo['extension'] == 'html' ||
+                        $pathinfo['extension'] == 'tpl' ||
+                        $pathinfo['basename'] == 'config.json'
+                    ) &&
+                    strpos($pathinfo['basename'], 'min.js') === false &&
+                    strpos($pathinfo['basename'], 'ext-') !== 0
+                ) ? true : false;
```

Comment:
It would be better to make a function for this to make it more understandable and readable.